lib/va_profile/models/email.rb

```diff
@@ -67,6 +67,17 @@ def self.build_from(body)
           va_profile_id: body['va_profile_id'] || body['vet360_id']
         )
       end
+
+      # Override the confirmation_date setter to correct it if it's after source_date.
+      # This prevents issues where client-provided dates may be ahead due to time differences.
+      # @param value [Time, String, nil] the confirmation date to set
+      # @return [Time, String, nil] the corrected confirmation date
+      def confirmation_date=(value)
```

Comment:
Wouldn't we need to update the `source_date` if `vets-api` is updating the data sent to VA Profile?